packages/eslint-plugin-turbo/lib/utils/calculate-inputs.ts

```diff
@@ -0,0 +1,449 @@
+
+  // Something like "build"
+  if (firstIndexOf === -1) {
+    return {
+      workspaceName: null,
+      scriptName: taskName,
+    };
+  }
+
+  // Something like "what#are#you#doing"
+  if (firstIndexOf !== taskName.lastIndexOf("#")) {
+    throw new Error("Invalid task name found in turbo.json.");
+  }
+
+  const [workspaceName, scriptName] = taskName.split("#");
+
+  return {
+    workspaceName,
+    scriptName,
+  };
+}
+
+export function getWorkspaceFromFilePath(
+  projectWorkspaces: WorkspaceConfig[],
+  filePath: string
+): WorkspaceConfig | null {
+  let possibleWorkspaces = projectWorkspaces
+    .filter((projectWorkspace) =>
+      filePath.startsWith(projectWorkspace.workspacePath)
+    )
+    .sort((a, b) => {
+      if (a.workspacePath > b.workspacePath) {
+        return -1;
+      } else if (a === b) {
+        return 0;
+      } else {
+        return 1;
+      }
+    });
+
+  if (possibleWorkspaces.length > 0) {
+    return possibleWorkspaces[0];
+  }
+
+  return null;
+}
+
+// Driver
+
+export class Project {
+  _key: ProjectKey;
+  _test: ProjectTests;
+
+  cwd: string | undefined;
+  allConfigs: WorkspaceConfig[];
+  projectRoot: WorkspaceConfig | undefined;
+  projectWorkspaces: WorkspaceConfig[];
+
+  constructor(cwd: string | undefined) {
+    this.cwd = cwd;
+    this.allConfigs = getWorkspaceConfigs(cwd);
+    this.projectRoot = this.allConfigs.find(
+      (workspaceConfig) => workspaceConfig.isWorkspaceRoot
+    );
+    this.projectWorkspaces = this.allConfigs.filter(
+      (workspaceConfig) => !workspaceConfig.isWorkspaceRoot
+    );
+
+    this._key = this.generateKey();
+    this._test = this.generateTestConfig();
+  }
+
+  valid(): boolean {
+    return this.allConfigs.length > 0;
+  }
+
+  generateKey(): ProjectKey {
+    let global: EnvironmentConfig = {
+      legacyConfig: [],
+      env: [],
+      passThroughEnv: null,
+      dotEnv: null,
+    };
+    let globalTasks: {
+      [script: string]: EnvironmentConfig;
+    } = {};
+    let workspaceTasks: {
+      [workspace: string]: {
+        [script: string]: EnvironmentConfig;
+      };
+    } = {};
+
+    if (
+      this.projectRoot &&
+      this.projectRoot.turboConfig &&
+      !("extends" in this.projectRoot)
+    ) {
+      const rootTurboJson = this.projectRoot;
+
+      global = processGlobal(
+        this.projectRoot.workspacePath,
+        this.projectRoot.turboConfig
+      );
+
+      Object.entries(this.projectRoot.turboConfig.pipeline).forEach(
+        ([taskName, taskDefinition]) => {
+          const { workspaceName, scriptName } = getTaskAddress(taskName);
+          if (workspaceName) {
+            workspaceTasks[workspaceName] = workspaceTasks[workspaceName] || {};
+            workspaceTasks[workspaceName][scriptName] = processTask(
+              rootTurboJson.workspacePath,
+              taskDefinition
+            );
+          } else {
+            globalTasks[scriptName] = processTask(
```

Comment:
This is something like `"build"` inside of the task key in the root `turbo.json`. These we apply to _all_ workspaces.